### PR TITLE
Reword de1726

### PR DESCRIPTION
**Pull request by alop**
Rewording commit message to denote that this is not in upstream

### CODE REVIEW

**Comment by mjpopow** [comment]
little PEP8 nitpick, can we fix the no newline at the EOF for test_live_migrate.py ?

**Comment by alop** [comment]
@mattpopow fixed newline

**Comment by mjpopow** [comment]
Thanks! One more question, what happens if the aggregate does not have the metadata set for ram_allocation_ratio? Do we need a try/catch around Line 141?

ram_ratio = float(aggr.get('metadetails')['ram_allocation_ratio'])

**Comment by kevinbringard** [comment]
Good question, @mattpopow. I imagine it'd just fail the ram check and the VM will fail to migrate, but I'm not 100% sure what specifically happens in python when you try to multiple by a null value.

@alop or @lwatta: Do we have any test coverage for this, and/or have we done functional testing to see what happens if it fails to get metadetails and/or ram_allocation_ratio?

**Comment by lwatta** [comment]
The target host has to be in an aggregate. If it's not then the migration will fail. Do I explicitly check for ram_allocation ratio? No that check is not present. We can add it but it seems like we explicitly force it in every aggregate.

File: nova/conductor/tasks/live_migrate.py [hunk 1]
```diff
@@ -106,13 +108,54 @@ def _check_destination_is_not_source(self):
...
+
     def _check_destination_has_enough_memory(self):
-        avail = self._get_compute_info(self.destination)['free_ram_mb']
+        instance_uuid = self.instance.uuid
+        dest = self.destination
+        aggr = self._get_aggregate_metadata(instance_uuid, dest)
+        # for icehouse replace aggr.metadetails with aggr.metadata.
+
+        # (schoksey replaced) ram_ratio = float(aggr.metadetails['ram_allocation_ratio'])
+        ram_ratio = float(aggr.get('metadetails')['ram_allocation_ratio'])
```
Comment (mjpopow) [inline nova/conductor/tasks/live_migrate.py]: @lwatta if getting ram_allocation_ratio fails here, will this cause the conductor to crash?

Comment (lwatta) [inline nova/conductor/tasks/live_migrate.py]: i won't know until i test it later. but it will probably throw an error

Comment (alop) [inline nova/conductor/tasks/live_migrate.py]: Where did this code originate from then? Has it been tested in the lab? Merging here will put this in for 1.0.13 release

Comment (lwatta) [inline nova/conductor/tasks/live_migrate.py]: It originated from Me and Sharmin
Yes. It was tested in SVL

Comment (lwatta) [inline nova/conductor/tasks/live_migrate.py]: I will test without ram_allocation set tonight. Most likely i will need to add a quick check. If there is no ram_allocation use default of 1.5

**Comment by lwatta** [comment]
I changed the code to check the aggregate for ram_allocation. If it doesn't find it it simply uses the default of 1.5. I tested it in SVL and seems to work fine.

**Comment by mjpopow** [comment]
:+1 good job on this one guys!

**Comment by alop** [comment]
@mattpopow i think you mean :+1:

**Comment by mjpopow** [comment]
:beers:

**Pull request merged**